go/client/cmd_ctl_watchdog_windows.go

```diff
@@ -50,44 +50,54 @@ func (c *CmdWatchdog) Run() (err error) {
 	//
 	// Note that we give up after 10 consecutive crashes
 
-	if c.G().SocketInfo == nil {
-		return errors.New("Uninitialized socket")
-	}
-	pipeName := c.G().SocketInfo.GetFile()
-	if len(pipeName) == 0 {
-		return errors.New("No pipe name")
-	}
-
 	countdown := c.restarts
 	for {
-		conn, err := npipe.DialTimeout(pipeName, time.Second*10)
-		if conn == nil {
-			// no service started. exit.
+		// Blocking wait on service. First, there has to be a pid
+		// file, because this is a forking command.
+		var fn string
+		if fn, err = c.G().Env.GetPidFile(); err != nil {
```

Comment:
I see what you are saying, but remember this is already a forking command - if the server wasn't up, it starts it and performs the blocking ping loop, so passing the pid through buys us little I think. Also, the service could be up already anyway. But it's not a huge task.